client/src/entrypoints/admin/userbar.js

```diff
@@ -281,35 +299,100 @@ class Userbar extends HTMLElement {
       return;
     }
 
-    const results = await axe.run(
-      {
-        exclude: {
-          fromShadowDom: ['wagtail-userbar'],
-        },
-      },
-      {
-        runOnly: {
-          type: 'rule',
-          values: ['empty-heading', 'heading-order', 'p-as-heading'],
-        },
-      },
-    );
+    const config = this.getAxeConfiguration();
+
+    if (!config) {
+      return;
+    }
+
+    const results = await axe.run(config.context, config.options);
 
-    // draft UI output for testing purposes
     if (results.violations.length) {
-      const axeCount = document.createElement('div');
-      axeCount.textContent = results.violations.length;
-      axeCount.classList.add('w-userbar-axe-count');
-      this.trigger.appendChild(axeCount);
+      const dialogtemplates = document.querySelectorAll(
+        '[data-wagtail-dialog]',
+      );
+      const dialogs = dialog(dialogtemplates, this.shadowRoot);
+
+      if (!dialogs.length) {
+        return;
+      }
+
+      const a11yErrorBadge = document.createElement('div');
+      const allyErrorsNumber = results.violations.reduce(
+        (sum, violation) => sum + violation.nodes.length,
+        0,
+      );
+      a11yErrorBadge.textContent = allyErrorsNumber;
+      a11yErrorBadge.classList.add('w-userbar-axe-count');
+      this.trigger.appendChild(a11yErrorBadge);
+
+      const modal = dialogs[0];
+      const accessibilityResultsBox = this.shadowRoot.querySelector(
+        '#accessibility-results',
+      );
+
+      const modalBody = modal.$el.querySelector('[data-dialog-body]');
+      const a11yRowTemplate = this.shadowRoot.querySelector(
+        '#w-a11y-result-row-template',
+      );
 
       const showAxeResults = () => {
-        results.violations.forEach((violation) => {
-          const annotation = document.createElement('div');
-          annotation.textContent = violation.description;
-          accessibilityTrigger.appendChild(annotation);
+        modal.show();
+        modalBody.innerHTML = '';
+
+        results.violations.forEach((violation, index) => {
```

Comment:
Is there a particular reason why we're clearing and repopulating the modal every time the modal is opened? Since `axe.run` is run only once, is it possible to populate the modal once outside of the `showAxeResults()` function and just call `modal.show()` when we want to show the results?

Reply:
Thanks. Have you added it somewhere?

Reply:
Sorry about that, just realised I had made the change on my local copy and forgot to commit it when merging. I’ll add this to our list of refactorings.